firmware/gameport-adapter/HidJoystick.h

```diff
@@ -34,7 +34,7 @@ class HidJoystick {
     m_subDescriptor = new HIDSubDescriptor{m_hidDescription.data, m_hidDescription.size};
     m_hidDevice.AppendDescriptor(m_subDescriptor);
 
-    log("Detected device: %s", joystick->getDescription().name);
+    DEBUG_LOG("Detected device: %s", joystick->getDescription().name);
```

Comment:
You mean by defining the log() function "volatile"?
The trick with the renaming of the log() function by
```
#ifndef NDEBUG
#define DEBUG_LOG(...) log(__VA_ARGS__)
```
will trigger the compilter to strip the whole log() function, when NDEBUG is defined.
This can be verified as the whole vsnprintf() function (<vfprintf>) is missing in the resulting ASM file.